File: alphabit-coding-challenge-03/a_little_bigger/submissions/accepted/alittlebitbigger.py
s = str(input())
def solution(w):
    i = len(w)
    run = True
    if len(set(list(w))) == 1 or "".join(sorted(list(w), reverse=True)) == w:
        return "kinda impossible"
    while run:
        x = sorted(list(w[i-2:]))
        t = max(loc for loc, val in enumerate(x) if val == w[i-2]) + 1
        if t == len(x):
            i -= 1
            continue
        ch = x.pop(t)
        tmp = w[:i-2]+ch+"".join(x)
        if tmp > w:
            w = tmp
            break
        else:
            i -= 1
        if i == 1:
            w = "kinda impossible"
            break
    return w
print(solution(s))

File: alphabit-coding-challenge-03/a_little_bigger/submissions/accepted/a_little_bit_bigger.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define max 100000

int main()
{
    char input[max];
    int flag;
    int i;
    
    //input*************************************

    fgets(input,100000,stdin);
    

    // traitment*************************************
    flag = 0;
    for (i = 1; i <strlen(input) ; i++)
    {
        if (input[strlen(input)-i]>input[strlen(input)-i-1]) {
            char tmp = input[strlen(input)-i];
            input[strlen(input)-i] = input[strlen(input)-i-1];
            input[strlen(input)-i-1] = tmp;
            i=strlen(input);
            flag = 1;
        }
    }
    // output*************************************
    
    if (flag)
    {
        for (i = 0; i < strlen(input); i++)
        {
            printf("%c",input[i]);
        }
    }else
    {
        printf("kinda impossible");
    }
    
    
    
    
    

    return 0;
}